services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/S3NativeClientConfiguration.java

```diff
@@ -87,11 +112,47 @@ public int maxConcurrency() {
         return maxConcurrency;
     }
 
+    public Executor futureCompletionExecutor() {
+        return futureCompletionExecutor;
+    }
+
+    /**
+     * Finalize which async executor service will be used for the created client. The default async executor
+     * service has at least 8 core threads and can scale up to at least 64 threads when needed depending
+     * on the number of processors available.
+     */
+    private Executor resolveAsyncFutureCompletionExecutor(ClientAsyncConfiguration config) {
+        Supplier<Executor> defaultExecutor = () -> {
```

Comment:
Cross-module APIs are protected APIs and we generally maintain backwards-compatiblity for protected APIs, so I didn't extract them. I don't anticipate updating this method very often. I think we've only updated it once since v2 was launched.

Reply:
Okay, makes sense. I still fear that one day someone will make a change to one of these Executors without knowing to update the other. Can we at least cross-reference them in the private Javadoc?